climada/util/plot.py

```diff
@@ -927,23 +949,32 @@ def subplots_from_gdf(
         print("Unknown subplot-title-generation function. Subplot titles will be column names.")
         title_subplots = lambda cols: [f"{col}" for col in cols]
 
-    # change default plot kwargs if plotting return periods
-    if colorbar_name.strip().startswith('Return Period'):
-        if 'cmap' not in kwargs.keys():
-            kwargs.update({'cmap': 'viridis_r'})
-        if 'norm' not in kwargs.keys():
-            kwargs.update(
-                {'norm': mpl.colors.LogNorm(
-                    vmin=gdf.values[:,1:].min(), vmax=gdf.values[:,1:].max()
-                    ),
-                'vmin': None, 'vmax': None}
-            )
+    # use log colorbar for return periods and impact
+    if (
+        colorbar_name.strip().startswith(('Return Period', 'Impact')) and
+        'norm' not in kwargs.keys() and
+        # check if there are no zeros values in gdf
+        np.sum(gdf_values == 0) == 0 and
```

Comment:
More explicit and less error-prone than summing booleans
```suggestion
        not np.any(gdf_values == 0)
```